api/v1/instancemanager.go

```diff
@@ -52,3 +52,7 @@ type ListHostsResponse struct {
 	// paging through out all the results.
 	NextPageToken string `json:"nextPageToken,omitempty"`
 }
+
+type Info struct {
```

Comment:
`Info` is quite generic in this context. From the implementation it seems this is specific to the Instance Manager configuration, perhaps `InstanceManagerConfig`. Let's add documentation to this `struct` and its fields as well.

Reply:
Considered `InstanceManagerConfig` as the more concrete name before, but it may be confused with `Config` in `pkg/app/instances/instances.go`.

I think that runtime is different from an instance manager as runtime is the system running cloud orchestrator, and instance manager is a component of cloud orchestrator.

Plus, `GET /info` was designed to return architecture or OS version which seem to be not configurable. Or is it okay to name Config something we cannot configure?
If so, how about `RuntimeConfig` to distinguish it from `InstanceManagerConfig`

Reply:
Oh, I see where the confusion is at.

There are two different concepts here whose values overlap: `Runtime` where the cloud orchestrator is deployed to, and `InstanceManagerType`, the implementation type used by the `cloud orchestartor` to perform instance management operations.

`Runtime` could have values like: `GAE (Google App Engine)` and `local`, whereas `InstanceManagerType` could have values like: `GCP (Google Cloud Platform)`, `AWS` and `local`. This mean that we could have a `local` deployment and do instance management in GCP for example.

I suggested `InstanceManagerConfig` as we were modifying the `InstanceManager` interface to provide this information. I'm not sure which concept fits your use case better here. In either case I think there's no need to  modify the `InstanceManager` interface as we can use the cloud orchetrator configuration to gather relevant information. The config is loaded here https://github.com/google/cloud-android-orchestration/blob/main/cmd/cloud_orchestrator/main.go#L162.

Reply:
> In either case I think there's no need to modify the InstanceManager interface as we can use the cloud orchetrator configuration to gather relevant information.

I see, thus made a new file to define the `Info` interface.

Reply:
I think it's fine to add the service information handling logic into the `app.go` file for now.

Reply:
Since cloud orchestrators with GCP/local instance managers should return different runtime types. is it okay to maintain current function locations?

Reply:
Right now we have something like:

```
func (m *GCEInstanceManager) GetInfo() (*apiv1.Info, error) {
	return &apiv1.Info{
		RuntimeType: apiv1.Cloud,
	}, nil
}
```

and this is not always true as  we could do instance management in GCP using a local deployment. 

Gathering **runtime** information ( information about the platform the `cloud orchestrator` has been deployed to) is common no matter the instance manager type that was in its configuration.

Reply:
I think I was confused 😅  
Then is it proper to get runtime type from `conf.toml`?  
I think it can be included as part of `type Config`.

Reply:
@ser-io PTAL

Reply:
Updated: now returns `config.Config` as source of data which is parsed result of `conf.toml`